lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js

```diff
@@ -0,0 +1,205 @@
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dims] - list of dimensions over which to perform a reduction
+* @param {Function} predicate - predicate function
+* @param {*} [thisArg] - predicate execution context
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {Error} second argument must be broadcast-compatible with the non-reduced dimensions of the input ndarray
+* @throws {TypeError} third argument must be an ndarray-like object
+* @throws {TypeError} options argument must be an object
+* @throws {Error} must provide valid options
+* @throws {TypeError} predicate must be a function
+* @returns {ndarray} output ndarray
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var ndarray = require( '@stdlib/ndarray/ctor' );
+* var isEven = require( '@stdlib/assert/is-even' ).isPrimitive;
+* var empty = require( '@stdlib/ndarray/empty' );
+*
+* // Create a data buffer:
+* var xbuf = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0 ] );
+*
+* // Define the shape of the input array:
+* var shape = [ 3, 1, 2 ];
+*
+* // Define the array strides:
+* var sx = [ 4, 4, 1 ];
+*
+* // Define the index offset:
+* var ox = 1;
+*
+* // Create an input ndarray:
+* var x = new ndarray( 'float64', xbuf, shape, sx, ox, 'row-major' );
+*
+* // Create an output ndarray:
+* var y = empty( [], {
+*     'dtype': 'bool'
+* });
+*
+* // Perform reduction:
+* var out = assign( x, 3, y, isEven );
+* // returns <ndarray>
+*
+* var v = out.get();
+* // returns true
+*/
+function assign( x, n, y, options, predicate, thisArg ) {
+	var nargs;
+	var opts;
+	var err;
+	var ord;
+	var ctx;
+	var cb;
+	var N;
+	var v;
+
+	if ( !isndarrayLike( x ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be an ndarray-like object. Value: `%s`.', x ) );
+	}
+	if ( !isndarrayLike( y ) ) {
+		throw new TypeError( format( 'invalid argument. Third argument must be an ndarray-like object. Value: `%s`.', y ) );
+	}
+
+	N = ndims( x );
+	nargs = arguments.length;
+
+	if ( nargs === 4 ) {
+		if ( !isFunction( options ) ) {
+			throw new TypeError( format( 'invalid argument. Predicate must be a function. Value: `%s`.', options ) );
+		}
+		cb = options;
+		ctx = thisArg;
+		opts = objectAssign( {}, defaults );
+		opts.dims = zeroTo( N );
+	} else if ( nargs === 5 ) {
+		if ( isFunction( options ) ) {
+			cb = options;
+			ctx = predicate;
+			opts = objectAssign( {}, defaults );
+			opts.dims = zeroTo( N );
+		}
+		if ( isObject( options ) ) {
+			if ( !isFunction( predicate ) ) {
+				throw new TypeError( format( 'invalid argument. Predicate must be a function. Value: `%s`.', predicate ) );
+			}
+			cb = predicate;
+			ctx = thisArg;
+			opts = objectAssign( {}, defaults, options );
+			err = validate( opts, N, options );
+			if ( err ) {
+				throw err;
+			}
+			if ( opts.dims === null ) {
+				opts.dims = zeroTo( N );
+			}
```

Comment:
Why does this need to be repeated here and again at L168. Again, think about simplifying code structure.